### PR TITLE
[file_crash_bug] Check the user status before adding a needinfo

### LINKED ISSUE #2128: [file_crash_bug] Don't needinfo users who are not active or blocks needinfos
See also: https://github.com/mozilla/bugbot/issues/2053 and https://github.com/mozilla/bugbot/issues/2043

### PR DESCRIPTION
**Pull request by suhaibmujahid**
Resolves #2128

## Checklist

- [x] Type annotations added to new functions
- [x] Docs added to functions touched in main classes
- [x] Dry-run produced the expected results
- [ ] The [`to-be-announced`](https://github.com/mozilla/bugbot/labels/to-be-announced) tag added if this is worth announcing

### CODE REVIEW

File: bugbot/rules/file_crash_bug.py [hunk 1]
```diff
@@ -113,13 +142,25 @@ def get_bugs(self, date):
...
                 "description": description,
-                # TODO: Uncomment the following lines when we move to file on
-                # the production instance of Bugzilla. Filling `regressed_by` or
-                # `flags` on bugzilla-dev will cause "bug does not exist" errors.
-                # "regressed_by": signature.regressed_by,
-                # "flags": flags,
             }
 
+            # Filling the `flags` field on bugzilla-dev will cause an error when
+            # the user is not exist.
```
Comment (marco-c) [inline bugbot/rules/file_crash_bug.py:148]:
```suggestion
            # the user does not exist.
```

**Review by marco-c — DISMISSED** [review]

File: bugbot/rules/file_crash_bug.py [hunk 2]
```diff
@@ -41,15 +43,48 @@ def get_bz_params(self, date):
             "keywords_type": "allwords",
         }
 
+    def _active_regression_authors(
+        self, signatures: list[SignatureAnalyzer]
+    ) -> set[str]:
+        """Get Bugzilla usernames for users who are active and can be needinfo'd.
+
+        Args:
+            signatures: signatures to check the status if their regression
```
Comment (marco-c) [inline bugbot/rules/file_crash_bug.py:52]:
```suggestion
            signatures: a list of signatures for which to check the status of their regression author
```

File: bugbot/rules/file_crash_bug.py [hunk 3]
```diff
@@ -63,24 +98,18 @@ def get_bugs(self, date):
...
-                # TODO: check user activity and if the ni? is open
-                flags = [
-                    {
-                        "name": "needinfo",
-                        "requestee": signature.regressed_by_author["name"],
-                        "status": "?",
-                        "new": "true",
-                    }
-                ]
+            # Weather we should needinfo the regression author.
```
Comment (marco-c) [inline bugbot/rules/file_crash_bug.py:101]:
```suggestion
            # Whether we should needinfo the regression author.
```

**Review by marco-c — DISMISSED** [review]

File: bugbot/rules/file_crash_bug.py (hunk 2 above)
Comment (suhaibmujahid) [reply bugbot/rules/file_crash_bug.py]: Done in 65ae5c269658bfdb2a9c28800c43a6a03fb5c425.

File: bugbot/rules/file_crash_bug.py (hunk 3 above)
Comment (suhaibmujahid) [reply bugbot/rules/file_crash_bug.py]: Done in 65ae5c269658bfdb2a9c28800c43a6a03fb5c425.

**Review by marco-c — APPROVED** [review]

**Pull request merged**